src/app/hooks/useFetchSearchImages.ts

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState } from 'react';
+import type { imagesResult } from '../../api/apis';
+import useFetch from './useFetch';
+
+export type fetchSearchImagesResult = {
+  imagesResult: imagesResult | null;
+  errorMessage: string | null;
+  isLoading: boolean;
+  isFetchingNewResult: boolean;
+};
+
+export default function useFetchSearchImages(
+  fetchMoreImages: boolean,
```

Comment:
Couldn't this be a function to fetch the next page instead of a boolean? That way, you wouldn't need to have a `useEffect` setting the value back to `false`. A function could be called once and should something go wrong it would return an error for example

Reply:
I did it this way because the scroll event fires a couple of times until the new images have been rendered (what takes up to 2 seconds). So that's the reason why I used a fetchMoreImages state on the Search page and set it to false immediately with the next render as well as use a nextPage state within the hook which is set +1 only when the current page fetch has been finished to allow the next fetch.

I think moving the page state to the search page would bring a lot of logic to the page.

I struggled a lot with infinite loops and multiple fetches here and made it work this way. But if you have a better idea...!? ;)

Reply:
Ah, of course, didn't think about that! Nevermind then, I don't have a better idea